lib/scala/semver/src/main/java/org/enso/semver/SemVer.java

```diff
@@ -0,0 +1,149 @@
+package org.enso.semver;
+
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import scala.util.Failure;
+import scala.util.Success;
+import scala.util.Try;
+
+/**
+ * A facade hiding the implementation of the Semantic Versioning used in Enso.
+ */
+public class SemVer {
+
+    private long major;
+    private long minor;
+    private long patch;
+    private Optional<String> preRelease;
```

Comment:
`Optional` field occupies 16B more than simple usage of `String` and non-`null`. Hopefully there aren't going to be millions of instances of this class.